lib/api/apiUtils/object/BackendInfo.js

```diff
@@ -11,12 +13,15 @@ class BackendInfo {
     * @param {string | undefined } bucketLocationConstraint - location
     * constraint for bucket based on bucket metadata
     * @param {string} requestEndpoint - endpoint to which request was made
+    * @param {string | undefined } legacyLocationConstraint - legacy location
+    * constraint
     */
     constructor(objectLocationConstraint, bucketLocationConstraint,
-        requestEndpoint) {
+        requestEndpoint, legacyLocationConstraint) {
```

Comment:
I feel like you don't technically need to pass it as an argument since you can just call BackendInfo.getLegacyLocationConstraint() again

Reply:
I guess it's mostly for performance, I don't want to call it in `controllingBackendParam()` and  every time I call `getControllingLocationConstraint()`